### ISSUE TITLE
Search project symbols not working on _python_ project

### ISSUE THREAD

**Issue by Tsirimaholy** [zed-industries/zed#16749]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

The project symbols search does not work when I'm on a python project when I update to **Zed 0.149.5**.

### Environment

Zed: v0.149.5 (Zed)
OS: Linux Wayland debian 6.2
Memory: 11.6 GiB
Architecture: x86_64
GPU: Intel(R) HD Graphics 5500 (BDW GT2) || Intel open-source Mesa driver || Mesa 22.3.6

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

2024-08-23T15:24:28.895340981+03:00 [ERROR] Unhandled method workspace/symbol
2024-08-23T15:24:29.259927959+03:00 [ERROR] Unhandled method workspace/symbol
2024-08-23T15:24:29.261055695+03:00 [ERROR] Unhandled method workspace/symbol
2024-08-23T15:24:29.466385276+03:00 [ERROR] Unhandled method workspace/symbol
2024-08-23T15:24:29.485074756+03:00 [ERROR] Unhandled method workspace/symbol
2024-08-23T15:24:29.926140834+03:00 [ERROR] Unhandled method workspace/symbol
2024-08-23T15:24:29.927463591+03:00 [ERROR] Unhandled method workspace/symbol
2024-08-23T15:24:30.089511803+03:00 [ERROR] Unhandled method workspace/symbol
2024-08-23T15:24:30.089643665+03:00 [ERROR] Unhandled method workspace/symbol
2024-08-23T15:24:30.25081833+03:00 [ERROR] Unhandled method workspace/symbol
2024-08-23T15:24:30.251634769+03:00 [ERROR] Unhandled method workspace/symbol
2024-08-23T15:24:30.649705298+03:00 [ERROR] Unhandled method workspace/symbol
2024-08-23T15:24:30.649822298+03:00 [ERROR] Unhandled method workspace/symbol
2024-08-23T15:24:30.773001296+03:00 [ERROR] Unhandled method workspace/symbol
2024-08-23T15:24:30.773081948+03:00 [ERROR] Unhandled method workspace/symbol
2024-08-23T15:24:30.920650152+03:00 [ERROR] Unhandled method workspace/symbol
2024-08-23T15:24:30.920848854+03:00 [ERROR] Unhandled method workspace/symbol
2024-08-23T15:24:31.482450742+03:00 [ERROR] Unhandled method workspace/symbol
2024-08-23T15:24:31.495343538+03:00 [ERROR] Unhandled method workspace/symbol
2024-08-23T15:24:31.732334966+03:00 [ERROR] Unhandled method workspace/symbol
2024-08-23T15:24:31.732450724+03:00 [ERROR] Unhandled method workspace/symbol
2024-08-23T15:24:31.882809744+03:00 [ERROR] Unhandled method workspace/symbol
2024-08-23T15:24:31.883773417+03:00 [ERROR] Unhandled method workspace/symbol
</pre></details>

**Comment by Tsirimaholy** [comment]
I think this issue is related to the Desktop Environment, KDE Wayland. I've tested it on other DE and it works correctly.
Same for  #16117.

**Comment by JosephTLyons** [comment]
I'm actually thinking this might be a deficiency within pyright itself.

**Comment by Tsirimaholy** [comment]
> I'm actually thinking this might be a deficiency within pyright itself.

That's possible, because it starts not working too on even on non KDE DE. And the log are the sames T.T.

**Comment by lvignoli** [comment]
@JosephTLyons turning off `ruff` alongside pyright makes workspace symbol search work!
See [there](https://github.com/zed-industries/zed/issues/15691#issuecomment-2346165813), and I was able to reproduce it locally.

So it seems to be a problem of colocation between the two LSPs of pyright and ruff.

Regarding project symbols search, I could not tell if they are provided by pyright or if Zed provides some stub implementation of it specifically for Python.

---
Specs:
Zed: v0.153.4 (Zed Preview)
OS: macOS 14.4.1
Memory: 16 GiB
Architecture: aarch64

**Comment by JosephTLyons** [comment]
I don't think Zed has anything to do with project symbol search, but we are responsible for the outline view of the current file, as we use tree sitter to capture those.

**Comment by skroth** [comment]
How does Zed decide which LSP to send the symbol search to? Presumably we would want it to go to pyright and not ruff in this case.

**Comment by lvignoli** [comment]
@skroth I enabled it in my config, so I manually removed it:

```diff
 "languages": {
    "Python": {
      "format_on_save": "on",
      "formatter": [
        { "code_actions": { "source.organizeImports.ruff": true } },
        { "language_server": { "name": "ruff" } }
      ],
-     "language_servers": ["pyright", "ruff"]
+     "language_servers": ["pyright"]
    },
}
```

**Comment by Tsirimaholy** [comment]
> @skroth I enabled it in my config, so I manually removed it:
> 
> ```diff
>  "languages": {
>     "Python": {
>       "format_on_save": "on",
>       "formatter": [
>         { "code_actions": { "source.organizeImports.ruff": true } },
>         { "language_server": { "name": "ruff" } }
>       ],
> -     "language_servers": ["pyright", "ruff"]
> +     "language_servers": ["pyright"]
>     },
> }
> ```

Yesss. I test that and it work. But the question is now how can i use pyright alongside ruff lsps.

**Comment by tino** [comment]
I still can't get this to work. With or without ruff. I see the requests going to pyright in the rpc call log, and I see answers coming back. It just doesn't show them in the modal...

**Comment by vultix** [comment]
I'm seeing the same issue - search project symbols doesn't work if the `ruff` language server is enabled. I'm using this as a workaround for now:
```json
  "languages": {
    "Python": {
      "language_servers": ["pyright", "!ruff"],
      "format_on_save": "on",
      "formatter":  {
        "external": {
          "command": ".venv/bin/ruff",
          "arguments": ["format", "--stdin-filename", "{buffer_path}"]
        }
      }
    }
  },
```

**Comment by mikayla-maki** [comment]
This issue seems to about mutli-LSP querying, e.g. if you have multiple LSPs configured, but they provide different subsets of features, we should fall back to the first LSP that answers the query.

**Comment by osiewicz** [comment]
Can y'all reproduce this? I can't with a provided repro, which makes me think it might've been fixed colaterally?

**Comment by Tsirimaholy** [comment]
> Can y'all reproduce this? I can't with a provided repro, which makes me think it might've been fixed colaterally?

Yeah! It works correctly on my side now. I think it is already fixed.
